editor/layout/index.js

```diff
@@ -28,29 +29,77 @@ import {
 	getNotices,
 } from '../selectors';
 
-function Layout( { mode, isSidebarOpened, notices, ...props } ) {
-	const className = classnames( 'editor-layout', {
-		'is-sidebar-opened': isSidebarOpened,
-	} );
-
-	return (
-		<div key="editor" className={ className }>
-			<DocumentTitle />
-			<NoticeList onRemove={ props.removeNotice } notices={ notices } />
-			<UnsavedChangesWarning />
-			<AutosaveMonitor />
-			<Header />
-			<div className="editor-layout__content">
-				<div className="editor-layout__editor">
-					{ mode === 'text' && <TextEditor /> }
-					{ mode === 'visual' && <VisualEditor /> }
+class Layout extends Component {
```

Comment:
Looking at the previous simple component, wonder if we should convert the region-focus into a HoC that could be used in different editor layouts.

Reply:
Do you think this should be a generic component in the `components` module? It looks like something that could be useful even outside an editor context.

Reply:
done in daa9cf3